### DISCUSSION TITLE
Better UI Idea

### DISCUSSION THREAD

**Discussion by Saradominist** [lllyasviel/Fooocus#906] [Ideas]
What if we removed the Styles tab since it's redundant if we choose to make it as a dropdown widget from gradio with multiselect and we can change aspect ratio as dropdown without multiselect
![Screenshot 2023-11-11 204254](https://github.com/lllyasviel/Fooocus/assets/53563800/d8fce081-a02e-4b5c-84d8-b7d45cc220be)
![image](https://github.com/lllyasviel/Fooocus/assets/53563800/d293e620-9913-4485-aaed-7a4021c1aa9d)

**Comment by Saradominist** [comment]
> What if we removed the Styles tab since it's redundant if we choose to make it as a dropdown widget from gradio with multiselect and we can change aspect ratio as dropdown without multiselect ![Screenshot 2023-11-11 204254](https://user-images.githubusercontent.com/53563800/282257770-d8fce081-a02e-4b5c-84d8-b7d45cc220be.png) ![image](https://user-images.githubusercontent.com/53563800/282257814-d293e620-9913-4485-aaed-7a4021c1aa9d.png)

I removed the Styles tab so it's all in the settings tab when first initially clicked on advanced checkbox

**Comment by lllyasviel** [comment]
The UI of Fooocus is not arbitrary decision – it is based on extensively conducted user experiments.

"Drop-down of Styles" discourages new user from trying different styles.

This is discussed previously:

*The side effect is that new users are significantly less likely to try new styles. In the user study we did at the scale of 7 individuals before the Fooocus release, for current Fooocus UI (tested with 4 individuals, all first-time users), all 4 users try at least one styles and see what happens (1 user tried two styles and 1 user tried a lot). However, if the UI became a drop-down (tested with another 3 individuals, all first-time users), all 3 users ignore this style selection (although 2 users still asked what is this, but 0 tried to change it).*

**Reply by Saradominist** [reply]
Fair enough I see your point. I probably could think of better ways while keeping in mind about user interaction

**Comment by Phrixus2023** [comment]
Losing Convenience by Obsessing Too Much on the UI

**Comment by 8Docent8** [comment]
My apologies for getting into the conversation. I am satisfied with the current interface. It is convenient and understandable and really provokes trying new styles, but sometimes it is quite difficult to find the right one among all the styles. It would not be bad to add the ability to select "Favorite" styles in the future. It can be a checkbox or a separate folder. And, of course, a small description of styles or even cataloging or "categories" so that you can divide styles into artistic, illustrated, realistic, etc. Thanks again for the work done and for Outpaint 2.6.

**Reply by AkoZ** [reply]
something has been done recently which is just putting UP your selected styles, then when you re-use , you'll see the "fav'" at top selections...

**Comment by lllyasviel** [comment]
perhaps add something to allow sorting and fav

**Comment by GoldMath** [comment]
A dynamic search bar with filtering could also help finding styles instead of scrolling a lot. 
Moving the active ones on top into an "Active" area with a separator would help manage, maybe lock that area "on top" too.

To push people to try them, thumbnails are a way to go as it's can be difficult to understand or memorize them all, but the list is already very long and adding these images would look even more massive, [Easy Diffusion](https://github.com/easydiffusion/easydiffusion) solution was [to categorize them](https://github.com/easydiffusion/easydiffusion/wiki/Image-Modifiers).

The [Fooocus-MRE/wiki](https://github.com/MoonRide303/Fooocus-MRE/wiki) been helpful with the styles thumbnail.

**Comment by tybiboune** [comment]
the ability to activate some "automatic styles selection according to prompt" could be great too :)
example: if you prompt some "cat made of paper" then it would activate all styles related to origami
type a "robotic" something or a "cyber" thingie, same...
use "dark atmosphere" or any combination or words that convey "darkness", switches on the "dark" style
etc

**Comment by Ambratolm** [comment]
Well I think it's good thing the styles are all visible without a dropdown, and I would suggest to enforce it more by giving more visually appealing elements to the styles list by adding thumbnails, style preview, descriptions, and group them by multiple similarities.
Close Example (Although I prefer to still give more space and size to text):
![Screenshot 2023-11-22 212342](https://github.com/lllyasviel/Fooocus/assets/27871399/51fef54e-c8b1-4f0b-8083-73cace3a8aca)

With the styles added in recent updates (♥ awesome) the list is going more and more long and convoluted, It really need badly some categorization with show/hide features.

![catstyl](https://github.com/lllyasviel/Fooocus/assets/27871399/0bc3f3aa-0d32-4c08-8647-6949d93bc1dc)

**Reply by Phrixus2023** [reply]
agree with you

**Reply by Saradominist** [reply]
How about something like this for less clutter? ironically I used Fooocus to make them
![Screenshot 2023-11-23 155231](https://github.com/lllyasviel/Fooocus/assets/53563800/c3edd008-6d1a-4655-82c2-1e1afe88f37d)

**Reply by Ambratolm** [reply]
good initiative, but the objective I'm aiming at, is to be able for example to know the expected result of "SAI Anime" right from the menu even if you don't know the meaning of the name, imagine if you don't know what Anime is, you may be thinking it'll give a realistic style. that's what happens with style names you're not familiar with, so i find myself most of the time playing with the ones having familiar names.

**Reply by Saradominist** [reply]
yeah I see where you're coming from. It's like a spoiler of what yets to come when the image is generated. So the user knows what to expect and not think the style is busted
